### PR TITLE
Move most of the `dart:ui` tests to `package:test`.

### PR DESCRIPTION
**Pull request by matanlurey**
Work towards https://github.com/flutter/flutter/issues/133569.

There are a couple with custom matchers I'd like to migrate in a follow-up PR.

### CODE REVIEW

File: testing/dart/color_test.dart [hunk 1]
```diff
@@ -244,9 +240,9 @@ void main() {
         alpha: 1, red: 1, green: 0, blue: 0, colorSpace: ColorSpace.displayP3);
     final Color srgb = p3.withValues(colorSpace: ColorSpace.extendedSRGB);
     expect(srgb.a, equals(1.0));
-    expect(srgb.r, approxEquals(1.0931));
-    expect(srgb.g, approxEquals(-0.22684034705162098));
-    expect(srgb.b, approxEquals(-0.15007957816123998));
+    expect(srgb.r, closeTo(1.0931, 1e4));
```
Comment (gaaclarke) [inline testing/dart/color_test.dart:243]: I think you meant `1e-4`?

**Review by gaaclarke — CHANGES_REQUESTED** [review]

File: testing/dart/color_test.dart (hunk 1 above)
Comment (matanlurey) [reply testing/dart/color_test.dart:243]: Yes, yes I did.

Comment (matanlurey) [reply testing/dart/color_test.dart]: Done.

**Review by gaaclarke — COMMENTED** [review]
lgtm modulo the delta values.  How did we end up with so many negative tests?  This is a big parenting no-no, "don't tell your kids what not to do, tell them what they should be doing." =)

**Review by gaaclarke — APPROVED** [review]
lgtm!

**Pull request merged**